panic when importing from directory

### Steps to reproduce

I compiled and ran warewulf on Ubuntu 22.04 from main as follows:

```
git clone https://github.com/warewulf/warewulf.git
apt install make unzip
snap install go --classic
cd warewulf
make all
export PREFIX=""
make install
systemctl start warewulfd
wwctl container import {dir}
```

### Error message

When running an import from directory:

```
root@ww1:~/warewulf# wwctl container import /ww/tmp/ubuntu-2204-nocuda --force                                                                                                                                                                                                                                            
Setting VNFS name: ubuntu-2204-nocuda                                                                                                                                                                                                                                                                                         
Overwriting existing VNFS                                                                                                                                                                                                                                                                                                     
ERRO[0000] Can't add file /ww/tmp/ubuntu-2204-nocuda/boot/System.map-5.15.0-107-generic to tar: io: read/write on closed pipe                                  
ERRO[0000] io: read/write on closed pipe                                                                                                                                                                                                                                                                                      
ERRO[0000] Can't close tar writer: io: read/write on closed pipe                                                                                                                                                                                                                                                              
panic: a library subroutine needed to run a subprocess, but reexec.Init() was not called in main()                                                                                                                                                                                                                            
                                                                                                                                                               
goroutine 1 [running]:                                                                                                                                                                                                                                                                                                        
github.com/containers/storage/pkg/reexec.panicIfNotInitialized(...)                                                                                                                                                                                                                                                           
        /root/warewulf/vendor/github.com/containers/storage/pkg/reexec/reexec.go:48                                                                                                                                                                                                                                           
github.com/containers/storage/pkg/reexec.Command({0xc000279b00?, 0x39?, 0xc000403940?})
        /root/warewulf/vendor/github.com/containers/storage/pkg/reexec/command_linux.go:21 +0x9d                                                               
github.com/containers/storage/pkg/chrootarchive.invokeUnpack({0x7f00a1641698, 0xc000279ad0}, {0xc000403940, 0x39}, 0xc000180240, {0xc000403940, 0x39})
        /root/warewulf/vendor/github.com/containers/storage/pkg/chrootarchive/archive_unix.go:89 +0x1d2         
github.com/containers/storage/pkg/chrootarchive.untarHandler({0x1297ea0, 0xc000224180}, {0xc000403940, 0x39}, 0x411e01?, 0x1, {0xc000403940, 0x39})            
        /root/warewulf/vendor/github.com/containers/storage/pkg/chrootarchive/archive.go:95 +0x355
github.com/containers/storage/pkg/chrootarchive.Untar({0x1297ea0?, 0xc000224180?}, {0xc000403940?, 0x10?}, 0xc0004eef10?)                                      
        /root/warewulf/vendor/github.com/containers/storage/pkg/chrootarchive/archive.go:36 +0x2e
github.com/containers/storage/pkg/archive.(*Archiver).TarUntar(0xc0000a8c80, {0x7ffcb1636838, 0x1a}, {0xc000403940, 0x39})                                     
        /root/warewulf/vendor/github.com/containers/storage/pkg/archive/archive.go:1200 +0x3f0                                                                                                                                                                                                                                
github.com/containers/storage/pkg/archive.(*Archiver).CopyWithTar(0xc0000a8c80, {0x7ffcb1636838, 0x1a}, {0xc000403940, 0x39})                                                                                                                                                                                                 
        /root/warewulf/vendor/github.com/containers/storage/pkg/archive/archive.go:1249 +0x1f2                                                                 
github.com/containers/storage/drivers/copy.DirCopy({0x7ffcb1636838, 0x1a}, {0xc000403940, 0x39}, 0x1a?, 0x98?)                                                                                                                                                                                                                
        /root/warewulf/vendor/github.com/containers/storage/drivers/copy/copy_unsupported.go:24 +0xa5                                                          
github.com/warewulf/warewulf/internal/pkg/container.ImportDirectory({0x7ffcb1636838, 0x1a}, {0x7ffcb1636840?, 0x0?})                                                                                                                                                                                                          
        /root/warewulf/internal/pkg/container/imprt.go:71 +0xf7
github.com/warewulf/warewulf/internal/pkg/api/container.ContainerImport(0xc0004efc88)
        /root/warewulf/internal/pkg/api/container/container.go:221 +0x745
github.com/warewulf/warewulf/internal/app/wwctl/container/imprt.CobraRunE(0xc00021a500?, {0xc0000a88e0?, 0x4?, 0x10daddd?})                                    
        /root/warewulf/internal/app/wwctl/container/imprt/main.go:27 +0xb4   
github.com/spf13/cobra.(*Command).execute(0x1ab2d60, {0xc0000a88c0, 0x2, 0x2})
        /root/warewulf/vendor/github.com/spf13/cobra/command.go:940 +0x882
github.com/spf13/cobra.(*Command).ExecuteC(0x1aaf100)  
        /root/warewulf/vendor/github.com/spf13/cobra/command.go:1068 +0x3a5                                                                                    
github.com/spf13/cobra.(*Command).Execute(0x101ad1360?)                   
        /root/warewulf/vendor/github.com/spf13/cobra/command.go:992 +0x13
main.main()                                             
        /root/warewulf/cmd/wwctl/main.go:13 +0x1e                       
root@ww1:~/warewulf#
```

### Information on your system

```root@ww1:~# cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
root@ww1:~# wwctl version
wwctl version:   4.5.0rc3+173.g88f0b1b5.dirty-1
rpc version: apiPrefix:"rc1" apiVersion:"1" warewulfVersion:"4.5.0rc3+173.g88f0b1b5.dirty-1"
root@ww1:~#```

### General information

- [X] I have run `wwctl version` and reported the contents of `/etc/os-release`
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have captured and reported relevant error messages and logs

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -56,6 +56,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Correctly extract smbios asset key during Grub boot. #1291
 - Refactor of `wwinit/init` to more properly address rootfs options. #1098
 - Fix autodetected kernel sorting issue. #1332
+- Avoid panic on container import #1244
 
 ## v4.5.7, 2024-09-11
 

--- a/internal/pkg/container/imprt.go
+++ b/internal/pkg/container/imprt.go
@@ -7,6 +7,7 @@ import (
 
 	"github.com/containers/image/v5/types"
 	"github.com/containers/storage/drivers/copy"
+	"github.com/containers/storage/pkg/reexec"
 	"github.com/pkg/errors"
 
 	warewulfconf "github.com/warewulf/warewulf/internal/pkg/config"
@@ -67,7 +68,9 @@ func ImportDirectory(uri string, name string) error {
 	if !util.IsFile(path.Join(uri, "/bin/sh")) {
 		return errors.New("Source directory has no /bin/sh: " + uri)
 	}
-
+	if reexec.Init() {
+		return errors.New("couldn't init reexec")
+	}
 	err = copy.DirCopy(uri, fullPath, copy.Content, true)
 	if err != nil {
 		return err

